### PR TITLE
[google-cloud-c] New port

### PR DESCRIPTION
**Pull request by SamuelMarks**
**Describe the pull request**
In cross-platform C89—with a libcurl wrapper—I wrote a Google Cloud Compute library.

- #### What does your PR fix?  
N/A

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
I am still working on this PR

Might not want to merge it for a-g-e-s, in order to give me time to write a code-generator to replace all the custom written `struct`s [+ `free` code] and API calls.

### CODE REVIEW

**Comment by dg0yt** [comment]
Wrong port name. It is not by Google.
And personally I think C89 out of scope for a modern C++ package manager.
And personally I think packages should have some relevance *before* being accepted into the main repo.

**Comment by SamuelMarks** [comment]
Yes well this is a draft PR. It'll probably be here a while to give me a chance to code-generate the bulk of the library from the official [Google Cloud Compute discovery JSON](https://www.googleapis.com/discovery/v1/apis/compute/v1/rest).

@dg0yt Wait, so you don't want C packages contributes to vcpkg?

**Comment by SamuelMarks** [comment]
Yes well this is a draft PR. It'll probably be here a while to give me a chance to code-generate the bulk of the library from the official [Google Cloud Compute discovery JSON](https://www.googleapis.com/discovery/v1/apis/compute/v1/rest).

@dg0yt Wait, so you don't want C packages contributed to vcpkg?

File: ports/google-cloud-c/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    SHA512          a940491a1590e61585c52815610a6bcb8ec2b41a8f6a270cf33b09b2053ee9bc8bb207a4b3bf02c24aecb51ae9bddf7cb5a29be427aeb69dcfafd6a68e06e235
+    HEAD_REF        master
+)
+
+vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}")
+vcpkg_cmake_install()
+file(INSTALL "${SOURCE_PATH}/cmake/License.txt"
+     DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
+     RENAME copyright)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
```
Comment (cenit) [inline ports/google-cloud-c/portfile.cmake:14]: sure about this line?

**Comment by cenit** [comment]
> @dg0yt Wait, so you don't want C packages contributed to vcpkg?

i don't think so.
i have yolo/darknet here and i think that it is not the least used port...

**Comment by dg0yt** [comment]
> @dg0yt Wait, so you don't want C packages contributed to vcpkg?

I didn't say anything about C in general. 
But I'm concerned about two many wrappers limiting the maintainability of the wrapped ports.
And I'm concerned about C standards from 30 years ago contributing to this problem.

**Comment by dg0yt** [comment]
This is a kindly reminder that this port shouldn't be prefixed with `google-` because it is not from Google (unlike port `google-cloud-cpp`).

**Comment by SamuelMarks** [comment]
@dg0yt I'm talking with Google out-of-band (email) and https://github.com/googleapis/google-cloud-cpp/issues/6522 about this project (once it's out of draft it could move to their org). Still a ways to go before out-of-draft (namely I'm writing a compiler to generate the API).

**Comment by LilyWangLL** [comment]
Pinging @SamuelMarks for response. Is work still being done for this PR?

**Comment by SamuelMarks** [comment]
Yes, work is still being done on this port.

**Comment by LilyWangLL** [comment]
Pinging @SamuelMarks for response. Is work still being done for this PR? There are no new changes for more than 1 month.

**Comment by SamuelMarks** [comment]
Yes, work is still being done on this port. I'm working specifically on https://github.com/SamuelMarks/cdd-c which will be used to generate the entire Google Cloud API hierarchy from their official discovery endpoint [in JSON] to C.

**Comment by LilyWangLL** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**